packages/widgetbook/lib/src/fields/date_time_field.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+
+import 'field.dart';
+import 'field_codec.dart';
+import 'field_type.dart';
+
+class DateTimeField extends Field<DateTime> {
+  DateTimeField({
+    required super.name,
+    super.initialValue,
+    super.onChanged,
+    required this.readOnly,
+  }) : super(
+          type: FieldType.dateTime,
+          codec: FieldCodec<DateTime>(
+            toParam: (value) {
+              final string = asString(value);
+              // replace all instances of ':' with '~' to avoid
+              // issues with retrieving the value from the param
+              // in the query group because it is saved in a map
+              return string.replaceAll(':', timeSeparator);
+            },
+            toValue: (param) {
+              if (param == null) return null;
+              final string = param.replaceAll(timeSeparator, ':');
+              return DateTime.tryParse(string);
+            },
```

Comment:
- Using `Uri.encodeComponent` and `Uri.decodeComponent` to convert the `:` into `%3A` would be a better approach than replacing it with a `~`. The URL encoding might be the default behaviour of all fields in the near future, that's why I prefer if we use it from now, to make this change easier in the future.
- The ISO 8601 date format will also result in a bad UX, so I prefer if we used something more simple as "yyyy-MM-dd hh:mm"

```suggestion
          codec: FieldCodec<DateTime>(
            toParam: (value) {
              return Uri.encodeComponent(
                '${value.year}-${value.month}-${value.day} '
                '${value.hour}:${value.minute}',
              );
            },
            toValue: (param) {
              return param == null
                  ? null
                  : DateTime.tryParse(
                      Uri.decodeComponent(param),
                    );
            },
          ),
```

Note that the following members should be removed with the new snippet:
1. `timeSeparator`
1. `_revertSeparator`
1. `asString`